Tools/build/deepfreeze.py

```diff
@@ -509,9 +510,20 @@ def main() -> None:
     args = parser.parse_args()
     verbose = args.verbose
     output = args.output
+
+    if args.file:
+        if verbose:
+            print(f"reading rules from {args.file}")
+        with open(args.file, "rt", encoding="utf-8") as fin:
+            rules = [x.strip() for x in fin.readlines()]
```

Comment:
```suggestion
        with open(args.file, "r", encoding="utf-8-sig") as fin:
            rules = [x.strip() for x in fin]
```

The 't' mode is unnecessary, the `utf-8-sig` encoding is slightly safer here (in case the file has a BOM), and `readlines()` is also unnecessary.

Reply:
Done. But the makefile routines explicitly provide the frozen module names, so I did the same by passing them as FrozenId properties.
One more thing, if I may. At first I didn't really understand how the compilation process handles the generated files, so this patch is really a mess.
If and when this PR is accepted, please squash-merge it. :-)

Thank you very much for the review.